### PR TITLE
Sys 3732 wireup offence pallet

### PR DESCRIPTION
**Pull request by ivan-cholakov**
Adding offences to eth-bridge, tests will be added in a follow-up pr

### CODE REVIEW

File: pallets/eth-bridge/src/offence.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    fn offenders(&self) -> Vec<Offender> {
+        self.offenders.clone()
+    }
+
+    fn session_index(&self) -> SessionIndex {
+        self.session_index
+    }
+
+    fn validator_set_count(&self) -> u32 {
```
Comment (alexpArtos) [inline pallets/eth-bridge/src/offence.rs:39]: Why do we need this? Isn't the result of this function to enable us to call `obj.validator_set_count` from the outside? But if that is the same name  of the internal field, why is this not redundant?

File: pallets/eth-bridge/src/offence.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+impl<Offender: Clone> Offence<Offender> for CorroborationOffence<Offender> {
+    const ID: Kind = *b"bridge:::offence";
+    type TimeSlot = SessionIndex;
+
+    fn offenders(&self) -> Vec<Offender> {
+        self.offenders.clone()
+    }
+
+    fn session_index(&self) -> SessionIndex {
```
Comment (alexpArtos) [inline pallets/eth-bridge/src/offence.rs:35]: see below

File: pallets/eth-bridge/src/tests/mock.rs [hunk 3]
```diff
@@ -2,20 +2,40 @@ use super::*;
...
+type Offence = crate::CorroborationOffence<IdentificationTuple>;
+pub struct OffenceHandler;
+impl ReportOffence<AccountId, IdentificationTuple, Offence> for OffenceHandler {
+    fn report_offence(reporters: Vec<AccountId>, offence: Offence) -> Result<(), OffenceError> {
+        OFFENCES.with(|l| l.borrow_mut().push((reporters, offence)));
+        Ok(())
+    }
+
+    fn is_known_offence(_offenders: &[IdentificationTuple], _time_slot: &SessionIndex) -> bool {
+        false
```
Comment (alexpArtos) [inline pallets/eth-bridge/src/tests/mock.rs:35]: Will this change later?

**Review by alexpArtos — APPROVED** [review]

File: pallets/eth-bridge/src/offence.rs (hunk 1 above)
Comment (ivan-cholakov) [reply pallets/eth-bridge/src/offence.rs:39]: this is the implementation of the `Offence` trait, so this implementation is just a stub, and yes it is redundant

File: pallets/eth-bridge/src/tests/mock.rs (hunk 3 above)
Comment (ivan-cholakov) [reply pallets/eth-bridge/src/tests/mock.rs:35]: this is just basic implementation, the tests are going to come with the `eth-bridge` unit tests pr

**Pull request merged**